conjure_oxide/tests/integration/basic/comprehension/01/input-expected-rule-trace-human.txt

```diff
@@ -0,0 +1,207 @@
+Model before rewriting:
+
+find m: matrix indexed by [[int(1..5)]] of int(1..5)
+find x: int(1..2)
+find y: int(1..2)
+
+such that
+
+(x = 1),
+and([((y < 5)) -> ((m[x] = x)) | x: int(1..5),(x < 4)])
```

Comment:
I am suprised the moving of y<5 to the head of the comprehension happens in the parser. The parser could parse it as is and a rule could move it to the head? Would that not be better?

Reply:
The precondition for this rule is whether the expression deeply contains any references. To check this is O(n): if this moving the guards rule, the check would have to be done in full on every node inside the comprehension to check if the rule applies.

Reply:
Memoization in metadata would allow for this sort of thing, but thats a way off yet..

Reply:
@Arnav-Bista